### PR TITLE
Make repeated builds return identical instances for BuiltList, BuiltMap, BuiltSet.

### PR DESCRIPTION
**Pull request by davidmorgan**

### CODE REVIEW

File: lib/src/list/list_builder.dart [hunk 1]
```diff
@@ -38,9 +38,11 @@ class ListBuilder<E> {
   /// The `ListBuilder` can be modified again and used to create any number
   /// of `BuiltList`s.
   BuiltList<E> build() {
-    if (_builtList != null) return _builtList;
-    _copyBeforeWrite = true;
```
Comment (dskloetg) [inline lib/src/list/list_builder.dart]: Below whenever you set _copyBeforeWrite, you also always need to set _builtList to null.

**Comment by BlackHC** [comment]
LGTM

File: lib/src/list/list_builder.dart (hunk 1 above)
Comment (davidmorgan) [inline lib/src/list/list_builder.dart]: Thanks, will send a separate pull request for that one.
